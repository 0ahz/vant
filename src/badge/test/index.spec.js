import Badge from '..';
import { mount } from '@vue/test-utils';

test('should render nothing when badge is empty string', () => {
  const wrapper = mount(Badge, {
    props: {
      badge: '',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should render nothing when badge is undefined', () => {
  const wrapper = mount(Badge, {
    props: {
      badge: undefined,
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should render nothing when badge is zero', () => {
  const wrapper = mount(Badge, {
    props: {
      badge: 0,
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should render content slot and match snapshot', () => {
  const wrapper = mount(Badge, {
    slots: {
      content: () => 'Custom Content',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});
