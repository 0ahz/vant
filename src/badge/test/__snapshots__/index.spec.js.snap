// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render content slot correctly 1`] = `
<div class="van-badge">
  Custom Content
</div>
`;

exports[`should render nothing when badge is empty string 1`] = ``;

exports[`should render nothing when badge is undefined 1`] = ``;

exports[`should render nothing when badge is zero 1`] = ``;
